support/agent/sim.go

```diff
@@ -0,0 +1,251 @@
+package agent
+
+import (
+	"context"
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin/reward"
+	"math"
+	big2 "math/big"
+	"math/rand"
+	"strings"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/cbor"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin"
+	"github.com/filecoin-project/specs-actors/v2/actors/builtin/power"
+	vm "github.com/filecoin-project/specs-actors/v2/support/vm"
+)
+
+type Sim struct {
+	Config   SimConfig
+	Accounts []address.Address
+	Miners   []*MinerAgent
+	v        *vm.VM
+	rnd      *rand.Rand
+}
+
+type SimConfig struct {
+	AccountCount           int
+	AccountInitialBalance  abi.TokenAmount
+	Seed                   int64
+	CreateMinerProbability float32
+}
+
+type ReturnHandler func(v *vm.VM, msg Message, ret cbor.Marshaler) error
+
+type Message struct {
+	From          address.Address
+	To            address.Address
+	Value         abi.TokenAmount
+	Method        abi.MethodNum
+	Params        interface{}
+	ReturnHandler ReturnHandler
+}
+
+type MinerPowerTable struct {
+	addr    address.Address
+	qaPower abi.StoragePower
+}
+
+type PowerTable struct {
+	blockReward  abi.TokenAmount
+	totalQAPower abi.StoragePower
+	minerPower   []MinerPowerTable
+}
```

Comment:
These structs confused me a lot when I first saw them. I mistook them for state, but I think they're just transient values. Perhaps push them down the file and don't export them.